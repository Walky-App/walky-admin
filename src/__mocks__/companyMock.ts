import { type ICompany } from '../interfaces/company'

export const companyMock: ICompany = {
  _id: '66ccbb8513c98d0f7012a44b',
  company_location_pin: [37.6782534, -120.9941554],
  company_name: 'Fake company for mocking purposes',
  company_address: '2608 McHenry Avenue, Modesto, CA, USA',
  company_city: 'Modesto',
  company_country: 'United States',
  company_dbas: ['Dank Flowers'],
  company_phone_number: '2223334444',
  company_state: 'CA',
  company_tax_id: '22-3333333',
  company_zip: '95350',
  created_by: 'donnie-client-2@hemptemps.com',
  facilities: ['66ccbd1513c98d0f7012a8f4'],
  users: ['66843da2c853cc8e1a1fc290'],
  createdAt: '2024-08-26T17:29:41.802Z',
  company_documents: [],
  payment_information: [],
  updatedAt: '2024-08-26T17:36:22.065Z',
}
