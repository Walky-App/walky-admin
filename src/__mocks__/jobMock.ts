import { type IUser } from '../interfaces/User'
import { type IFacility } from '../interfaces/facility'
import { type IJob, type IJobShiftDay, type IRestriction, type IFeedback, type IDnr } from '../interfaces/job'

export const jobMock: IJob = {
  _id: '66bf8bc2907a582955d1b0c1',
  uid: '2408711',
  created_by: 'shannon@hemptemps.com',
  facility: {} as IFacility,
  title: 'Trimmer',
  start_time: '2024-08-16T12:30:00.000+00:00',
  end_time: '2024-08-16T21:00:00.000+00:00',
  total_hours: 8,
  lunch_break: 30,
  job_days: [] as IJobShiftDay[],
  job_dates: [
    '2024-08-19T06:00:00.000+00:00',
    '2024-08-20T06:00:00.000+00:00',
    '2024-08-21T06:00:00.000+00:00',
    '2024-08-22T06:00:00.000+00:00',
    '2024-08-23T06:00:00.000+00:00',
  ],
  job_tips: ['Parking Onsite', 'Parking on Street'],
  vacancy: 10,
  hourly_rate: 19,
  applicants: [
    {
      user: {} as IUser,
      is_active: false,
      is_working: false,
      _id: '66bf9e8f2fe241f8e7b433e7',
      first_name: 'John',
      last_name: 'Doe',
      rejection_reason: '',
    },
    {
      user: {} as IUser,
      is_active: false,
      is_working: false,
      _id: '66bf9f012fe241f8e7b43edd',
      first_name: 'Jane',
      last_name: 'Doe',
      rejection_reason: 'Some weird reason',
    },
  ],
  saved_by: [],
  dnr: [] as IDnr[],
  is_completed: true,
  is_full: false,
  is_active: true,
  feedback: [] as IFeedback[],
  clock_in_restriction: [] as IRestriction[],
  clock_out_restriction: [] as IRestriction[],
  distance: 0,
  createdAt: new Date('2024-08-16T17:26:26.729+00:00'),
  updatedAt: new Date('2024-08-24T06:00:09.550+00:00'),
  applicants_feedback_ids: [],
}
