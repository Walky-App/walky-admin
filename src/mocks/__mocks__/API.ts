// __mocks__/API.ts
const API = {
    post: jest.fn(), // we'll customize this in individual tests
  };
  
  export default API;
  