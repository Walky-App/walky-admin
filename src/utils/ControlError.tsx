import type { FieldErrors } from 'react-hook-form'

export function getFormErrorMessage(path: string, errors: FieldErrors) {
  const pathParts = path.split('.')
  let error: FieldErrors = errors

  for (const part of pathParts) {
    if (typeof error !== 'object' || error === null) {
      return null
    }
    error = error[part as keyof typeof error] as FieldErrors
  }

  if (error?.message) {
    return error.message ? <p className="mt-2 text-sm text-red-600">{String(error.message)}</p> : null
  }

  return null
}
