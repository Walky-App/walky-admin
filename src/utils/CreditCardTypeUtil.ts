export const getCardType = (number: string) => {
  const firstDigit = number[0]
  const firstTwoDigits = number.slice(0, 2)
  const firstFourDigits = number.slice(0, 4)

  if (/^4/.test(firstDigit)) {
    return 'Visa'
  } else if (/^5[1-5]/.test(firstTwoDigits)) {
    return 'MasterCard'
  } else if (/^3[47]/.test(firstTwoDigits)) {
    return 'Amex'
  } else if (/^6(?:011|5)/.test(firstFourDigits)) {
    return 'Discover'
  } else {
    return 'Unknown'
  }
}
