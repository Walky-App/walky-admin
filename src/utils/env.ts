export const getEnv = () => {
  return {
    VITE_API_BASE_URL: import.meta.env.VITE_API_BASE_URL || 'https://staging.walkyapp.com/api'
  };
};
