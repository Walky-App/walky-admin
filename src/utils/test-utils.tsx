import React, { type ReactElement } from 'react'

import { MemoryRouter } from 'react-router-dom'

import '@testing-library/jest-dom'
import { render, type RenderOptions } from '@testing-library/react'

const AllTheProviders = ({ children }: { children: React.ReactNode }) => {
  const originalConsoleError = console.error
  const jsDomCssError = 'Error: Could not parse CSS stylesheet'
  console.error = (...params) => {
    if (!params.find(p => p.toString().includes(jsDomCssError))) {
      originalConsoleError(...params)
    }
  }
  return <MemoryRouter>{children}</MemoryRouter>
}

const customRender = (ui: ReactElement, options?: Omit<RenderOptions, 'wrapper'>) =>
  render(ui, { wrapper: AllTheProviders, ...options })

export * from '@testing-library/react'
export { customRender as render }
