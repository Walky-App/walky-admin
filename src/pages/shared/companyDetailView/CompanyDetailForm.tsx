import { useState, useEffect } from 'react'

import { useForm, type SubmitHandler, Controller } from 'react-hook-form'
import { useParams, useNavigate } from 'react-router-dom'

import { format } from 'date-fns'
import { Button } from 'primereact/button'
import { ConfirmPopup, confirmPopup } from 'primereact/confirmpopup'
import { InputMask } from 'primereact/inputmask'
import { InputText } from 'primereact/inputtext'
import { MultiSelect, type MultiSelectChangeEvent } from 'primereact/multiselect'
import { classNames } from 'primereact/utils'

import { AddressAutoComplete } from '../../../components/shared/forms/AddressAutoComplete'
import { type IAddressAutoComplete } from '../../../components/shared/forms/AddressAutoComplete'
import { HtInputHelpText } from '../../../components/shared/forms/HtInputHelpText'
import { HtInputLabel } from '../../../components/shared/forms/HtInputLabel'
import { HtInfoTooltip } from '../../../components/shared/general/HtInfoTooltip'
import { type IUser } from '../../../interfaces/User'
import { type ICompany } from '../../../interfaces/company'
import { type IFacility } from '../../../interfaces/facility'
import { requestService } from '../../../services/requestServiceNew'
import { useUtils } from '../../../store/useUtils'
import { getFormErrorMessage } from '../../../utils/formUtils'
import { requiredFieldsNoticeText } from '../../../utils/formUtils'
import { roleChecker } from '../../../utils/roleChecker'

interface ICompanyFormInputs {
  company_name: string
  company_dbas?: string[]
  company_tax_id: string
  company_address: string
  company_phone_number: string
  company_city: string
  company_state: string
  company_zip: string
  company_country: string
  facilities: string[] | IFacility[]
  users: string[]
}

export const CompanyDetailForm = ({ selectedCompanyData }: { selectedCompanyData: ICompany }) => {
  const [moreAddressDetails, setMoreAddressDetails] = useState<IAddressAutoComplete | undefined>()
  const params = useParams()
  const navigate = useNavigate()
  const { showToast } = useUtils()
  const role = roleChecker()

  const [allFacilities, setFacilities] = useState<IFacility[]>([])
  const [allClients, setClients] = useState<IUser[]>([])

  useEffect(() => {
    const getAllFacilities = async () => {
      try {
        let response
        if (role === 'client') {
          response = await requestService({ path: `facilities/company/${params.id}` })
        } else {
          response = await requestService({ path: `facilities/company/${params.id}/with-unassigned-facilities` })
        }

        if (response.ok) {
          const fetchedFacilities: IFacility[] = await response.json()
          setFacilities(fetchedFacilities)
        } else {
          setFacilities([])
        }
      } catch (error) {
        console.error('Error fetching facilities data:', error)
      }
    }

    const getAllClients = async () => {
      try {
        const response = await requestService({ path: 'users/clients' })
        if (response.ok) {
          const fetchedClients: IUser[] = await response.json()
          setClients(fetchedClients)
        } else {
          setClients([])
        }
      } catch (error) {
        console.error('Error fetching clients data:', error)
      }
    }

    getAllFacilities()
    getAllClients()
  }, [params, role])

  const defaultCompanyFormValues: ICompanyFormInputs = {
    company_name: '',
    company_dbas: [],
    company_tax_id: '',
    company_address: '',
    company_phone_number: '',
    company_city: '',
    company_state: '',
    company_zip: '',
    company_country: '',
    facilities: [],
    users: [],
  }

  const {
    control,
    formState: { errors },
    handleSubmit,
    setValue,
  } = useForm<ICompanyFormInputs>({
    defaultValues: defaultCompanyFormValues,
  })

  useEffect(() => {
    if (moreAddressDetails) {
      if (moreAddressDetails.zip != null) {
        setValue('company_zip', moreAddressDetails.zip)
      }
      if (moreAddressDetails.state != null) {
        setValue('company_state', moreAddressDetails.state)
      }
      if (moreAddressDetails.city != null) {
        setValue('company_city', moreAddressDetails.city)
      }
      if (moreAddressDetails.address != null) {
        setValue('company_address', moreAddressDetails.address)
      }
      if (moreAddressDetails.country != null) {
        setValue('company_country', moreAddressDetails.country)
      }

      setMoreAddressDetails(undefined)
    }
  }, [moreAddressDetails, setValue])

  useEffect(() => {
    if (selectedCompanyData) {
      setValue('company_name', selectedCompanyData.company_name)
      setValue('company_dbas', selectedCompanyData.company_dbas)
      setValue('company_address', selectedCompanyData.company_address)
      setValue('company_phone_number', selectedCompanyData.company_phone_number)
      setValue('company_city', selectedCompanyData.company_city)
      setValue('company_state', selectedCompanyData.company_state)
      setValue('company_zip', selectedCompanyData.company_zip)
      setValue('company_country', selectedCompanyData.company_country)
      setValue('facilities', selectedCompanyData.facilities)
      setValue('users', selectedCompanyData.users)
    }

    if (selectedCompanyData.company_tax_id != null) {
      setValue('company_tax_id', selectedCompanyData.company_tax_id)
    }
  }, [selectedCompanyData, setValue])

  const onSubmit: SubmitHandler<ICompanyFormInputs> = async data => {
    try {
      const response = await requestService({
        path: `companies/${params.id}/with-facility-assignment`,
        method: 'PATCH',
        body: JSON.stringify(data),
      })
      if (!response.ok) {
        throw new Error('Failed to update company')
      }
      showToast({ severity: 'success', summary: 'Company updated successfully' })
    } catch (error) {
      console.error('Error updating company: ', error)
      showToast({ severity: 'error', summary: 'Failed to update company' })
    }
  }

  const accept = async () => {
    try {
      const response = await requestService({ path: `companies/${selectedCompanyData._id}`, method: 'DELETE' })

      if (response.ok) {
        const data = await response.json()
        showToast({ severity: 'success', summary: 'Success', detail: data.message })
        navigate('/admin/companies')
      }
    } catch (error) {
      console.error(error)
      showToast({ severity: 'error', summary: 'Error', detail: 'Error deleting facility' })
      navigate('/admin/companies')
    }
  }

  const reject = () => {
    showToast({ severity: 'warn', summary: 'Rejected', detail: 'You have canceled the facility delete' })
  }

  const handleDeleteConfirm = (event: React.MouseEvent<HTMLButtonElement>) => {
    event.preventDefault()
    confirmPopup({
      target: event.currentTarget as HTMLElement,
      message:
        'If you delete this Company, all associated facilities, USERS, jobs, licenses, images and company association will be removed. Are you sure you want to proceed?',
      icon: 'pi pi-info-circle',
      defaultFocus: 'reject',
      acceptClassName: 'p-button-danger',
      accept,
      reject,
    })
  }

  return (
    <form onSubmit={handleSubmit(onSubmit)} className="p-fluid">
      <div className="space-y-4 sm:space-y-12">
        <div className="grid grid-cols-1 gap-x-8 py-12 sm:gap-y-10 md:grid-cols-3">
          <div>
            <h2 className="text-base font-semibold leading-7 text-gray-900">Company Information</h2>
            <p className="mt-1 text-sm leading-6 text-gray-600">
              Please take a moment to provide the essential information for the company.
              {requiredFieldsNoticeText}
              <br />
              <br />
              Created by: {selectedCompanyData.created_by} on{' '}
              {format(selectedCompanyData?.createdAt ?? new Date(), 'MM/dd/yyyy')}
            </p>
          </div>

          <div className="grid max-w-2xl grid-cols-1 gap-x-6 gap-y-6 sm:grid-cols-6 md:col-span-2">
            <div className="sm:col-span-3">
              <Controller
                control={control}
                name="company_name"
                rules={{ required: 'Company Name is required' }}
                render={({ field, fieldState }) => (
                  <>
                    <HtInfoTooltip message="A corporate name is the legal name of a corporation.">
                      <HtInputLabel htmlFor={field.name} asterisk labelText="Company Name" />
                    </HtInfoTooltip>
                    <InputText
                      id={field.name}
                      {...field}
                      className={classNames({ 'p-invalid': fieldState.invalid }, 'mt-2')}
                    />
                    {getFormErrorMessage(field.name, errors)}
                  </>
                )}
              />
            </div>

            <div className="sm:col-span-3">
              <Controller
                control={control}
                name="company_dbas"
                render={({ field, fieldState }) => (
                  <>
                    <HtInfoTooltip message="Doing Business As (DBA) is also known as a trade name, fictitious business name, or assumed business name.">
                      <HtInputLabel htmlFor={field.name} labelText="Company DBAs" />
                    </HtInfoTooltip>
                    <InputText
                      id={field.name}
                      value={field.value?.join(', ')}
                      onChange={e => field.onChange(e.target.value.split(', '))}
                      className={classNames({ 'p-invalid': fieldState.invalid }, 'mt-2')}
                      aria-describedby={`${field.name}-help`}
                    />
                    <HtInputHelpText
                      fieldName={field.name}
                      helpText="Enter DBAs separated by a comma: dba1, dba2, dba3"
                    />
                    {getFormErrorMessage(field.name, errors)}
                  </>
                )}
              />
            </div>

            <div className="sm:col-span-3">
              <Controller
                control={control}
                name="company_tax_id"
                render={({ field, fieldState }) => (
                  <>
                    <HtInfoTooltip message="A Tax Identification Number (TIN) in the United States is a unique identifier assigned to individuals and businesses for tax purposes.">
                      <HtInputLabel htmlFor={field.name} labelText="Tax ID" />
                    </HtInfoTooltip>
                    <InputMask
                      id={field.name}
                      {...field}
                      mask="99-9999999"
                      slotChar="x"
                      className={classNames({ 'p-invalid': fieldState.invalid }, 'mt-2')}
                      autoComplete="off"
                    />
                    {getFormErrorMessage(field.name, errors)}
                  </>
                )}
              />
            </div>

            <div className="sm:col-span-3">
              <Controller
                control={control}
                name="company_phone_number"
                rules={{
                  required: 'Phone Number is required, should be 10 digits.',
                }}
                render={({ field, fieldState }) => (
                  <>
                    <HtInfoTooltip message="The phone number of the company. This is the number that administrators would call if they have questions or need to contact you.">
                      <HtInputLabel htmlFor={field.name} asterisk labelText="Company Phone Number" />
                    </HtInfoTooltip>
                    <InputMask
                      id={field.name}
                      {...field}
                      mask="(999) 999-9999"
                      slotChar="x"
                      unmask={true}
                      className={classNames({ 'p-invalid': fieldState.invalid }, 'mt-2')}
                      autoComplete="off"
                    />
                    {getFormErrorMessage(field.name, errors)}
                  </>
                )}
              />
            </div>

            <div className="sm:col-span-6">
              <Controller
                control={control}
                name="company_address"
                rules={{ required: 'Address is required' }}
                render={({ field, fieldState }) => (
                  <>
                    <HtInfoTooltip message="Legal address of the company.">
                      <HtInputLabel htmlFor={field.name} asterisk labelText="Company Address" />
                    </HtInfoTooltip>
                    <AddressAutoComplete
                      controlled
                      setMoreAddressDetails={setMoreAddressDetails}
                      currentAddress={field.value}
                      onChange={field.onChange}
                      value={field.value}
                      classNames={classNames({ 'p-invalid': fieldState.invalid }, 'mt-2')}
                      aria-describedby={`${field.name}-help`}
                    />
                    <HtInputHelpText fieldName={field.name} helpText="Only Commercial Address" />
                    {getFormErrorMessage(field.name, errors)}
                  </>
                )}
              />
            </div>

            {role === 'admin' ? (
              <div className="sm:col-span-6">
                <Controller
                  control={control}
                  name="users"
                  render={({ field, fieldState }) => {
                    const selectedClients = field.value
                      .map(id => allClients.find(client => client._id === id))
                      .filter(Boolean)
                    return (
                      <>
                        <HtInfoTooltip message="All clients related to this company">
                          <HtInputLabel htmlFor={field.name} labelText="Users" />
                        </HtInfoTooltip>
                        <MultiSelect
                          id={field.name}
                          {...field}
                          value={selectedClients}
                          optionLabel="email"
                          options={allClients}
                          display="chip"
                          selectAll
                          selectAllLabel="Select All"
                          onChange={(e: MultiSelectChangeEvent) => {
                            field.onChange(e.value.map((client: IUser) => client._id))
                          }}
                          placeholder="Select Services"
                          className={classNames({ 'p-invalid': fieldState.invalid }, 'mt-2')}
                        />
                        <HtInputHelpText
                          fieldName={field.name}
                          helpText="Please select all users related to this company."
                        />
                        {getFormErrorMessage(field.name, errors)}
                      </>
                    )
                  }}
                />
              </div>
            ) : null}

            <div className="sm:col-span-6">
              <Controller
                control={control}
                name="facilities"
                render={({ field, fieldState }) => {
                  const selectedFacilities = field.value
                    .map(item => allFacilities.find(facility => facility._id === item))
                    .filter(Boolean)
                  return (
                    <>
                      <HtInfoTooltip message="All facilities related to this company">
                        <HtInputLabel htmlFor={field.name} labelText="Facilities" />
                      </HtInfoTooltip>
                      <MultiSelect
                        id={field.name}
                        {...field}
                        value={selectedFacilities}
                        optionLabel="name"
                        options={allFacilities}
                        display="chip"
                        selectAll
                        selectAllLabel="Select All"
                        onChange={(e: MultiSelectChangeEvent) => {
                          field.onChange(e.value.map((facility: IFacility) => facility._id))
                        }}
                        placeholder="Select Services"
                        className={classNames({ 'p-invalid': fieldState.invalid }, 'mt-2')}
                        disabled={role === 'client'}
                      />
                      <HtInputHelpText fieldName={field.name} helpText="All facilities which this company owns." />
                      {getFormErrorMessage(field.name, errors)}
                    </>
                  )
                }}
              />
            </div>
            <div className="flex gap-2 sm:col-span-6">
              <Button
                type="submit"
                label="Update Company"
                icon="pi pi-save"
                pt={{ label: { className: 'text-nowrap' } }}
              />
              {role === 'admin' ? (
                <div className="basis-2/5">
                  <ConfirmPopup />
                  <Button
                    onClick={handleDeleteConfirm}
                    icon="pi pi-times"
                    label="Delete Company"
                    severity="danger"
                    pt={{ label: { className: 'text-nowrap' } }}
                  />
                </div>
              ) : null}
            </div>
          </div>
        </div>
      </div>
    </form>
  )
}
