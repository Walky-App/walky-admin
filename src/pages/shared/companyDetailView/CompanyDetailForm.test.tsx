import { companyMock } from '../../../__mocks__/companyMock'
import { render } from '../../../utils/test-utils'
import { CompanyDetailForm } from './CompanyDetailForm'

describe('CompanyDetailForm', () => {
  it('should render company detail form', () => {
    render(<CompanyDetailForm selectedCompanyData={companyMock} />)
  })

  it('should match snapshot', () => {
    const { asFragment } = render(<CompanyDetailForm selectedCompanyData={companyMock} />)
    expect(asFragment()).toMatchSnapshot()
  })
})
