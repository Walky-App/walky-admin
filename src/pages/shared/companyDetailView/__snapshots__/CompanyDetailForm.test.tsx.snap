// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CompanyDetailForm should match snapshot 1`] = `
<DocumentFragment>
  <form
    class="p-fluid"
  >
    <div
      class="space-y-4 sm:space-y-12"
    >
      <div
        class="grid grid-cols-1 gap-x-8 py-12 sm:gap-y-10 md:grid-cols-3"
      >
        <div>
          <h2
            class="text-base font-semibold leading-7 text-gray-900"
          >
            Company Information
          </h2>
          <p
            class="mt-1 text-sm leading-6 text-gray-600"
          >
            Please take a moment to provide the essential information for the company.
          </p>
          <p
            class="mt-1 text-sm leading-6 text-gray-600"
          >
            <span
              style="color: red;"
            >
              *
            </span>
             indicates a required field.
          </p>
          <br />
          <br />
          Created by: donnie-client-2@hemptemps.com on 08/26/2024
          <p />
        </div>
        <div
          class="grid max-w-2xl grid-cols-1 gap-x-6 gap-y-6 sm:grid-cols-6 md:col-span-2"
        >
          <div
            class="sm:col-span-3"
          >
            <div
              class="flex flex-row items-center gap-x-2"
            >
              <i
                class="custom-target-icon pi pi-info-circle p-text-secondary p-overlay-badge cursor-pointer text-lg lg:text-sm"
                data-pr-at="left top-5"
                data-pr-classname="text-sm text-wrap"
                data-pr-my="left bottom"
                data-pr-position="top"
                data-pr-tooltip="A corporate name is the legal name of a corporation."
              />
              <label
                class="block text-sm font-medium leading-6"
                for="company_name"
              >
                <span
                  style="color: red;"
                >
                  * 
                </span>
                Company Name
              </label>
            </div>
            <input
              class="mt-2 p-inputtext p-component p-filled"
              data-pc-name="inputtext"
              data-pc-section="root"
              id="company_name"
              name="company_name"
              value="Fake company for mocking purposes"
            />
          </div>
          <div
            class="sm:col-span-3"
          >
            <div
              class="flex flex-row items-center gap-x-2"
            >
              <i
                class="custom-target-icon pi pi-info-circle p-text-secondary p-overlay-badge cursor-pointer text-lg lg:text-sm"
                data-pr-at="left top-5"
                data-pr-classname="text-sm text-wrap"
                data-pr-my="left bottom"
                data-pr-position="top"
                data-pr-tooltip="Doing Business As (DBA) is also known as a trade name, fictitious business name, or assumed business name."
              />
              <label
                class="block text-sm font-medium leading-6"
                for="company_dbas"
              >
                Company DBAs
              </label>
            </div>
            <input
              aria-describedby="company_dbas-help"
              class="mt-2 p-inputtext p-component p-filled"
              data-pc-name="inputtext"
              data-pc-section="root"
              id="company_dbas"
              value="Dank Flowers"
            />
            <small
              id="company_dbas-help"
            >
              Enter DBAs separated by a comma: dba1, dba2, dba3
            </small>
          </div>
          <div
            class="sm:col-span-3"
          >
            <div
              class="flex flex-row items-center gap-x-2"
            >
              <i
                class="custom-target-icon pi pi-info-circle p-text-secondary p-overlay-badge cursor-pointer text-lg lg:text-sm"
                data-pr-at="left top-5"
                data-pr-classname="text-sm text-wrap"
                data-pr-my="left bottom"
                data-pr-position="top"
                data-pr-tooltip="A Tax Identification Number (TIN) in the United States is a unique identifier assigned to individuals and businesses for tax purposes."
              />
              <label
                class="block text-sm font-medium leading-6"
                for="company_tax_id"
              >
                Tax ID
              </label>
            </div>
            <input
              autocomplete="off"
              class="mt-2 p-inputmask p-inputtext p-component p-filled"
              data-pc-name="inputtext"
              data-pc-section="root"
              id="company_tax_id"
              name="company_tax_id"
              type="text"
            />
          </div>
          <div
            class="sm:col-span-3"
          >
            <div
              class="flex flex-row items-center gap-x-2"
            >
              <i
                class="custom-target-icon pi pi-info-circle p-text-secondary p-overlay-badge cursor-pointer text-lg lg:text-sm"
                data-pr-at="left top-5"
                data-pr-classname="text-sm text-wrap"
                data-pr-my="left bottom"
                data-pr-position="top"
                data-pr-tooltip="The phone number of the company. This is the number that administrators would call if they have questions or need to contact you."
              />
              <label
                class="block text-sm font-medium leading-6"
                for="company_phone_number"
              >
                <span
                  style="color: red;"
                >
                  * 
                </span>
                Company Phone Number
              </label>
            </div>
            <input
              autocomplete="off"
              class="mt-2 p-inputmask p-inputtext p-component p-filled"
              data-pc-name="inputtext"
              data-pc-section="root"
              id="company_phone_number"
              name="company_phone_number"
              type="text"
            />
          </div>
          <div
            class="sm:col-span-6"
          >
            <div
              class="flex flex-row items-center gap-x-2"
            >
              <i
                class="custom-target-icon pi pi-info-circle p-text-secondary p-overlay-badge cursor-pointer text-lg lg:text-sm"
                data-pr-at="left top-5"
                data-pr-classname="text-sm text-wrap"
                data-pr-my="left bottom"
                data-pr-position="top"
                data-pr-tooltip="Legal address of the company."
              />
              <label
                class="block text-sm font-medium leading-6"
                for="company_address"
              >
                <span
                  style="color: red;"
                >
                  * 
                </span>
                Company Address
              </label>
            </div>
            <span
              autocomplete="off"
              class="mt-2 p-autocomplete p-component p-inputwrapper p-inputwrapper-filled"
              data-pc-name="autocomplete"
              data-pc-section="root"
              id="pr_id_2"
            >
              <input
                aria-autocomplete="list"
                aria-expanded="false"
                aria-haspopup="listbox"
                autocomplete="off"
                class="p-autocomplete-input p-inputtext p-component p-filled"
                data-pc-name="inputtext"
                data-pc-section="root"
                name="address"
                placeholder="2608 McHenry Avenue, Modesto, CA, USA"
                role="combobox"
                type="text"
                value="2608 McHenry Avenue, Modesto, CA, USA"
              />
            </span>
            <small
              id="company_address-help"
            >
              Only Commercial Address
            </small>
          </div>
          <div
            class="sm:col-span-6"
          >
            <div
              class="flex flex-row items-center gap-x-2"
            >
              <i
                class="custom-target-icon pi pi-info-circle p-text-secondary p-overlay-badge cursor-pointer text-lg lg:text-sm"
                data-pr-at="left top-5"
                data-pr-classname="text-sm text-wrap"
                data-pr-my="left bottom"
                data-pr-position="top"
                data-pr-tooltip="All facilities related to this company"
              />
              <label
                class="block text-sm font-medium leading-6"
                for="facilities"
              >
                Facilities
              </label>
            </div>
            <div
              class="mt-2 p-multiselect p-component p-inputwrapper p-multiselect-chip"
              data-pc-name="multiselect"
              data-pc-section="root"
              id="facilities"
            >
              <div
                class="p-hidden-accessible"
                data-p-hidden-accessible="true"
                data-pc-section="hiddeninputwrapper"
              >
                <input
                  aria-expanded="false"
                  data-pc-section="input"
                  name="facilities"
                  readonly=""
                  role="combobox"
                  tabindex="0"
                  type="text"
                />
              </div>
              <div
                class="p-multiselect-label-container"
                data-pc-section="labelcontainer"
              >
                <div
                  class="p-multiselect-label p-placeholder"
                  data-pc-section="label"
                >
                  Select Services
                </div>
              </div>
              <div
                class="p-multiselect-trigger"
                data-pc-section="trigger"
              >
                <svg
                  aria-hidden="true"
                  class="p-icon p-multiselect-trigger-icon p-c"
                  data-pc-section="triggericon"
                  fill="none"
                  height="14"
                  viewBox="0 0 14 14"
                  width="14"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M7.01744 10.398C6.91269 10.3985 6.8089 10.378 6.71215 10.3379C6.61541 10.2977 6.52766 10.2386 6.45405 10.1641L1.13907 4.84913C1.03306 4.69404 0.985221 4.5065 1.00399 4.31958C1.02276 4.13266 1.10693 3.95838 1.24166 3.82747C1.37639 3.69655 1.55301 3.61742 1.74039 3.60402C1.92777 3.59062 2.11386 3.64382 2.26584 3.75424L7.01744 8.47394L11.769 3.75424C11.9189 3.65709 12.097 3.61306 12.2748 3.62921C12.4527 3.64535 12.6199 3.72073 12.7498 3.84328C12.8797 3.96582 12.9647 4.12842 12.9912 4.30502C13.0177 4.48162 12.9841 4.662 12.8958 4.81724L7.58083 10.1322C7.50996 10.2125 7.42344 10.2775 7.32656 10.3232C7.22968 10.3689 7.12449 10.3944 7.01744 10.398Z"
                    fill="currentColor"
                  />
                </svg>
              </div>
            </div>
            <small
              id="facilities-help"
            >
              All facilities which this company owns.
            </small>
          </div>
          <div
            class="flex gap-2 sm:col-span-6"
          >
            <button
              aria-label="Update Company"
              class="p-button p-component"
              data-pc-name="button"
              data-pc-section="root"
              type="submit"
            >
              <span
                class="p-button-icon p-c p-button-icon-left pi pi-save"
                data-pc-section="icon"
              />
              <span
                class="p-button-label p-c text-nowrap"
                data-pc-section="label"
              >
                Update Company
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </form>
</DocumentFragment>
`;
